Project2015To2017/Analysis/Diagnostics/W001IllegalProjectTypeDiagnostic.cs

```diff
@@ -0,0 +1,54 @@
+using Project2015To2017.Definition;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Project2015To2017.Analysis.Diagnostics
+{
+	public sealed class W001IllegalProjectTypeDiagnostic : DiagnosticBase
+	{
+		/// <inheritdoc />
+		public W001IllegalProjectTypeDiagnostic(Project project = null, AnalysisOptions options = null) : base(1, project, options)
```

Comment:
We need to use a constant for these analyzer ids so we can have a descriptive ‘skip diagnostics’ array that has the actual types that we opt out of.